applications/DEM_application/custom_elements/rigid_body_element.cpp

```diff
@@ -1,100 +1,347 @@
     : Element(NewId, ThisNodes) {
         mpIntegrationScheme = NULL;
     }
     
-    Element::Pointer RigidBodyElement::Create(IndexType NewId, NodesArrayType const& ThisNodes, PropertiesType::Pointer pProperties) const {
-        return Element::Pointer(new RigidBodyElement(NewId, GetGeometry().Create(ThisNodes), pProperties));
+    Element::Pointer RigidBodyElement3D::Create(IndexType NewId, NodesArrayType const& ThisNodes, PropertiesType::Pointer pProperties) const {
+        return Element::Pointer(new RigidBodyElement3D(NewId, GetGeometry().Create(ThisNodes), pProperties));
     }      
     
     // Destructor
-    RigidBodyElement::~RigidBodyElement() {
-
+    RigidBodyElement3D::~RigidBodyElement3D() {
+    
+        mListOfCoordinates.clear();  
+        if (mpIntegrationScheme != NULL) delete mpIntegrationScheme;
+        // Destroy triangles?
     }
       
-    void RigidBodyElement::Initialize(ProcessInfo& r_process_info) {
-
+    void RigidBodyElement3D::Initialize(ProcessInfo& r_process_info, ModelPart& rigid_body_element_sub_model_part) {
+        
+        GetGeometry()[0].Set(DEMFlags::FIXED_VEL_X, false);
+        GetGeometry()[0].Set(DEMFlags::FIXED_VEL_Y, false);
+        GetGeometry()[0].Set(DEMFlags::FIXED_VEL_Z, false);
+        GetGeometry()[0].Set(DEMFlags::FIXED_ANG_VEL_X, false);
+        GetGeometry()[0].Set(DEMFlags::FIXED_ANG_VEL_Y, false);
+        GetGeometry()[0].Set(DEMFlags::FIXED_ANG_VEL_Z, false);
+              
+        mInertias = ZeroVector(3);
+        
+        mInertias[0] = rigid_body_element_sub_model_part[RIGID_BODY_INERTIAS][0];
+        mInertias[1] = rigid_body_element_sub_model_part[RIGID_BODY_INERTIAS][1];
+        mInertias[2] = rigid_body_element_sub_model_part[RIGID_BODY_INERTIAS][2];
+        mMass = rigid_body_element_sub_model_part[RIGID_BODY_MASS]; 
+        
+        GetGeometry()[0].FastGetSolutionStepValue(ORIENTATION) = Quaternion<double>(1.0, 0.0, 0.0, 0.0);
+        GetGeometry()[0].FastGetSolutionStepValue(NODAL_MASS) = mMass;    
+        GetGeometry()[0].FastGetSolutionStepValue(EXTERNAL_APPLIED_FORCE)[0] = rigid_body_element_sub_model_part[EXTERNAL_APPLIED_FORCE][0];
+        GetGeometry()[0].FastGetSolutionStepValue(EXTERNAL_APPLIED_FORCE)[1] = rigid_body_element_sub_model_part[EXTERNAL_APPLIED_FORCE][1];
+        GetGeometry()[0].FastGetSolutionStepValue(EXTERNAL_APPLIED_FORCE)[2] = rigid_body_element_sub_model_part[EXTERNAL_APPLIED_FORCE][2];
+        GetGeometry()[0].FastGetSolutionStepValue(EXTERNAL_APPLIED_MOMENT)[0] = rigid_body_element_sub_model_part[EXTERNAL_APPLIED_MOMENT][0];
+        GetGeometry()[0].FastGetSolutionStepValue(EXTERNAL_APPLIED_MOMENT)[1] = rigid_body_element_sub_model_part[EXTERNAL_APPLIED_MOMENT][1];
+        GetGeometry()[0].FastGetSolutionStepValue(EXTERNAL_APPLIED_MOMENT)[2] = rigid_body_element_sub_model_part[EXTERNAL_APPLIED_MOMENT][2];
+        
+        mFloatingFlag = rigid_body_element_sub_model_part[FLOATING_OPTION];
```

Comment:
Not part of a generic RigidBody. It should be in a derived class which can interact with water or something like this... I think @GuillermoCasas said this already.